bird_cloud_gnn/radar_dataset.py

```diff
@@ -152,19 +163,32 @@ def process(self):
 
         self.graphs = []
         self.labels = np.array([])
-        for data_file in os.listdir(self.data_folder):
-            self._read_one_file(data_file)
+        if self.data_folder is not None:
+            if os.path.isdir(self.data_folder):
+                for data_file in os.listdir(self.data_folder):
+                    self._read_one_file(os.path.join(self.data_folder, data_file))
+            else:
+                if not os.path.isfile(self.data_folder):
+                    raise ValueError("`data_folder` is neither a file nor a directory")
+                self._read_one_file(self.data_folder)
+
+        else:
+            if not isinstance(self.input_data, pd.DataFrame):
+                raise ValueError(
+                    "if `self.data_folder` is not set, `self.input_data` should be a pandas.DataFrame"
+                )
+            self._process_data(self.input_data)
```

Comment:
Flattening again. Also, I don't think we should check for instances during processing.
```suggestion
        elif self.input_data is not None:
            self._process_data(self.input_data)
        else:
            raise ValueError("Missing input. Either self.data_folder or self.input_data needs to be defined.")
```